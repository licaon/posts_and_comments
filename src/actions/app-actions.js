import { INIT } from 'src/constants/action-types';

export const initApp = (payload = {}) => ({
  type: INIT,
  payload,
});
