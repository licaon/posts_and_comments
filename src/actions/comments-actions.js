import { GET_COMMENTS, ADD_COMMENTS, TOGGLE_COMMENTS } from 'src/constants/action-types';

export const getComments = (payload = {}) => ({
  type: GET_COMMENTS,
  payload,
});

export const addComments = (payload = []) => ({
  type: ADD_COMMENTS,
  payload,
});

export const toggleCommentsState = (payload = {}) => ({
  type: TOGGLE_COMMENTS,
  payload,
});
