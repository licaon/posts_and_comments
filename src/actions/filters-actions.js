import { ADD_SEARCH_FILTER, ADD_SELECT_FILTER } from 'src/constants/action-types';

export const addSearchFilter = (payload = {}) => ({
  type: ADD_SEARCH_FILTER,
  payload,
});

export const addSelectFilter = (payload = {}) => ({
  type: ADD_SELECT_FILTER,
  payload,
});
