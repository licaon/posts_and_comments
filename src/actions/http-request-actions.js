import { START_LOADING_DATA, END_LOADING_DATA } from 'src/constants/action-types';

export const startLoadingData = (payload = {}) => ({
  type: START_LOADING_DATA,
  payload,
});

export const doneLoadingData = (payload = {}) => ({
  type: END_LOADING_DATA,
  payload,
});
