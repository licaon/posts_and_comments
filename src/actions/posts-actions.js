import { GET_POSTS, ADD_POST, REQUEST_POST_DELETE, DELETE_POST } from 'src/constants/action-types';

export const getPosts = (payload = {}) => ({
  type: GET_POSTS,
  payload,
});

export const addPost = (payload = []) => ({
  type: ADD_POST,
  payload,
});

export const requestPostDelete = (payload = {}) => ({
  type: REQUEST_POST_DELETE,
  payload,
});

export const confirmPostDelete = (payload = {}) => ({
  type: DELETE_POST,
  payload,
});
