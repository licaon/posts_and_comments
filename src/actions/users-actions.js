import { GET_USERS, ADD_USER } from 'src/constants/action-types';

export const getUsers = (payload = {}) => ({
  type: GET_USERS,
  payload,
});

export const addUser = (payload = []) => ({
  type: ADD_USER,
  payload,
});
