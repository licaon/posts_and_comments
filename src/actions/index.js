import * as HttpRequestActions from 'src/actions/http-request-actions';
import * as AppActions from 'src/actions/app-actions';
import * as PostsActions from 'src/actions/posts-actions';
import * as FiltersActions from 'src/actions/filters-actions';

export {
  HttpRequestActions,
  AppActions,
  PostsActions,
  FiltersActions,
};
