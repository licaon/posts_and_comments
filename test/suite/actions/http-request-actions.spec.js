import * as HttpRequestActions from 'src/actions/http-request-actions';

describe('HttpRequestActions', () => {
  it('should execute action correctly with argument - #startLoadingData', (done) => {
    expect(HttpRequestActions.startLoadingData({ test: 'test' })).toMatchSnapshot();
    done();
  });
  it('should execute action correctly without argument - #startLoadingData', (done) => {
    expect(HttpRequestActions.startLoadingData()).toMatchSnapshot();
    done();
  });
  it('should execute action correctly with argument - #doneLoadingData', (done) => {
    expect(HttpRequestActions.doneLoadingData({ test: 'test' })).toMatchSnapshot();
    done();
  });
  it('should execute action correctly without argument - #doneLoadingData', (done) => {
    expect(HttpRequestActions.doneLoadingData()).toMatchSnapshot();
    done();
  });
});
