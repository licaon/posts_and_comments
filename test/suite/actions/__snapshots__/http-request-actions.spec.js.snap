// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HttpRequestActions should execute action correctly with argument - #doneLoadingData 1`] = `
Object {
  "payload": Object {
    "test": "test",
  },
  "type": "END_LOADING_DATA",
}
`;

exports[`HttpRequestActions should execute action correctly with argument - #startLoadingData 1`] = `
Object {
  "payload": Object {
    "test": "test",
  },
  "type": "START_LOADING_DATA",
}
`;

exports[`HttpRequestActions should execute action correctly without argument - #doneLoadingData 1`] = `
Object {
  "payload": Object {},
  "type": "END_LOADING_DATA",
}
`;

exports[`HttpRequestActions should execute action correctly without argument - #startLoadingData 1`] = `
Object {
  "payload": Object {},
  "type": "START_LOADING_DATA",
}
`;
